### DISCUSSION TITLE
Disable and re-enable fade in a backlog does not work sometimes

### DISCUSSION THREAD

**Discussion by Th0maz** [arendst/Tasmota#16326] [Q&A] [answered]
Hey everybody,
on my tasmotized ceiling light I configured fade 1 and speed 5 to get some smooth light changes.
I also use some rules to signal certain events and therefore would like to disable fade.
Setoption34 is set to 0 to have no delay in rule backlogs.

When I try:
Rule1 on Event#Brightness do backlog fade 0; dimmer 100; fade 1 endon
fade is sometimes disabled when the event occurs and sometimes it's not.

I tried the same with speed setting instead of fade. same result.

Any ideas on this?

**Comment by sfromis** [comment]
The Fade and Speed commands are not meant to be used during a command sequence, as they do not cause actions, they are settings to be configured going forward.

**Reply by Th0maz** [reply]
Yes, I got this.
Usually I have fade enabled and speed set to 5 for every change of this light.
For this special case I would like to disable fading temporarily, change e.g. the brightness, and re-enable fading right after this.

**Reply by sfromis** [reply]
Not a supported way of using it. You might be able to do something with changes while power is off, not tested this.

**Reply by sfromis** [reply]
You also have the command [Speed2](https://tasmota.github.io/docs/Commands/#speed2) which can be feasible for some type of temporary steps, like possibly a very fast speed for a step change fade....

**Reply by s-hadinger** [reply]
This is strange. There is a special `Speed2 0` command that sets the Fade/Speed only for the next light command. This should work in your case. Documentation is not explicit enough. See https://github.com/arendst/Tasmota/pull/15954

**Comment by Th0maz** [comment, marked as answer]
Thank you guys. speed2 0 and speed2 ! does exactly what I was looking for :)
